### PR TITLE
My hw 05

### PR DESCRIPTION
**Pull request by AlexeyShparun**
Add my hw 05. Includes specs for 3 and 4 hw

### CODE REVIEW

File: HW03/alexey_shparun/spec/hw_03_task_03_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          2018-12-31 23:59:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+          2018-12-31 23:59:49.8 ubuntu-xenial[14319] Debug - docker event processed
+          2019-01-01 00:00:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
+          2019-01-01 00:00:38.8 ubuntu-xenial[14319] Debug - Calling core with action: messages
+          2019-01-01 00:00:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+          2019-01-01 00:00:59.8 ubuntu-xenial[14319] Debug - inside docker_handle_event
+        LOGS
+      end
+
+      it 'return type of strind' do
```
Comment (alexshgov) [inline HW03/alexey_shparun/spec/hw_03_task_03_spec.rb:22]:
```suggestion
      it 'return type of string do
```

Comment (AlexeyShparun) [reply HW03/alexey_shparun/spec/hw_03_task_03_spec.rb]: corrected, you can check

File: HW04/alexey_shparun/spec/homework_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  describe 'homework with the wrong parameters' do
+    context 'when no arguments given' do
+      it 'returns ArgumentError' do
+        expect { described_class.new }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'when not all arguments have been passed on' do
+      it 'not all arguments' do
```
[the remark this answers is not in the document]
Comment (AlexeyShparun) [reply HW04/alexey_shparun/spec/homework_spec.rb]: corrected, you can check it

File: HW04/alexey_shparun/spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  end
+
+  describe 'mark_as_read!' do
+    context 'when student marks as read' do
+      before do
+        student.to_work!(hw)
+      end
+
+      it 'return empty array' do
+        expect { student.mark_as_read! }.to change(student, :notifications_list).to eq []
```
[the remark this answers is not in the document]
Comment (AlexeyShparun) [reply HW04/alexey_shparun/spec/student_spec.rb]: corrected, you can check

File: HW04/alexey_shparun/spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    context 'when no arguments given' do
+      it 'returns ArgumentError' do
+        expect { student.to_check! }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  describe 'mark_as_read!' do
```
[the remark this answers is not in the document]
Comment (AlexeyShparun) [reply HW04/alexey_shparun/spec/student_spec.rb]: corrected, you can check

File: HW04/alexey_shparun/spec/student_spec.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    context 'when no arguments given' do
+      it 'returns ArgumentError' do
+        expect { student.add_answer! }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  describe 'to_work!' do
```
[the remark this answers is not in the document]
Comment (AlexeyShparun) [reply HW04/alexey_shparun/spec/student_spec.rb]: corrected, you can check it

File: HW04/alexey_shparun/spec/student_spec.rb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+require_relative '../lib/notification'
+
+RSpec.describe Student do
+  let(:title) { 'HW04' }
+  let(:description) { 'description' }
+  let(:student) { described_class.new(name: 'Alexey', surname: 'Shparun') }
+  let(:mentor) { Mentor.new(name: 'Vasya', surname: 'Klapan') }
+  let(:hw) { Homework.new(title, description, student, mentor) }
+
+  describe 'add_answer!' do
```
[the remark this answers is not in the document]
Comment (AlexeyShparun) [reply HW04/alexey_shparun/spec/student_spec.rb]: corrected, you can check it

File: HW04/alexey_shparun/spec/mentor_spec.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+require_relative '../lib/mentor'
+require_relative '../lib/student'
+require_relative '../lib/notification'
+
+RSpec.describe Mentor do
+  let(:student) { Student.new(name: 'Alexey', surname: 'Shparun') }
+  let(:mentor) { described_class.new(name: 'Vasya', surname: 'Klapan') }
+  let(:hw) { mentor.add_homework(title: 'HW04', description: 'description', student: student) }
+
+  describe 'add_homework' do
```
[the remark this answers is not in the document]
Comment (AlexeyShparun) [reply HW04/alexey_shparun/spec/mentor_spec.rb]: corrected, you can check it

**Review by alexshagov — APPROVED** [review]

**Pull request merged**